src/mame/brother/lw30.cpp

```diff
@@ -0,0 +1,1105 @@
+// license:BSD-3-Clause
+// copyright-holders:Bartman/Abyss
+
+#include "emu.h"
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+#include "machine/timer.h"
+#include "cpu/z180/z180.h"
+#include "imagedev/floppy.h"
+#include "util/utf8.h"
+
+#include "sound/beep.h"
+#include "video/mc6845.h"
+
+#define LOG_FLOPPY (1U << 1)
+#define LOGFLOPPY(...) LOGMASKED(LOG_FLOPPY, __VA_ARGS__)
+
+//#define VERBOSE (LOG_GENERAL | LOG_FLOPPY)
+//#define VERBOSE (LOG_GENERAL)
+#include "logmacro.h"
+
+// if updating project, c:\users\chuck\downloads\msys64\win32env.bat
+// cd \schreibmaschine\mame_src
+// make SUBTARGET=brother NO_USE_MIDI=1 NO_USE_PORTAUDIO=1 vs2019
+
+// command line parameters:
+// -log -debug -window -intscalex 2 -intscaley 2 lw30 -resolution 960x256 -flop roms\lw30\tetris.img
+// -debug -autoboot_script c:\schreibmaschine\mame_src\lw30.lua -log -window -intscalex 8 -intscaley 8 -resolution 1920x512 lw30 -flop c:\brothers\mame\disks\lw30\tetris.img
+
+// floppy: see src\devices\bus\vtech\memexp\floppy.cpp
+
+//////////////////////////////////////////////////////////////////////////
+// LW-30
+//////////////////////////////////////////////////////////////////////////
+
+// *** Hit Ctrl+Q during bootup to be able to start programs (like Tetris) from disk
+
+/***************************************************************************
+
+Brother LW-30
+1991
+
+Hardware:
+
+#7 
+Hitachi
+HD64180RP6
+8-bit CMOS Micro Processing Unit
+fully compatible with Zilog Z80180 (Z180)
+6 MHz, DP-64S, Address Space 512 K Byte
+MuRata CST12MTW 12.00 MHz Ceramic Resonator
+
+#8 
+Mitsubishi
+M65122ASP
+UA5445-B LC-1
+
+#6 
+NEC
+D23C4001EC-172
+UA2849-A
+4MBit Mask ROM for Dictionary
+
+#5
+LH532H07
+UA5362-A
+2MBit Mask ROM
+
+#11 
+Hitachi
+HM6264ALP-15
+High Speed CMOS Static RAM (8kbit x 8) 150ns
+
+#10
+Mitsubishi
+M65017FP
+UA5498-A
+Murata CST4MGW 4.00 MHz Ceramic Resonator
+
+#3, #4
+Mitsubishi
+HD74LS157P
+
+#1, #2
+NEC
+D41464C-10
+Dynamic NMOS RAM (64kbit x 4) 100ns
+
+QA1, QA2
+Mitsubishi
+M54587P
+
+#12
+Texas Instruments
+SN74HC04N
+
+Floppy - custom single sided 3.5" DD
+240kb capacity
+256 bytes/sector
+12 sectors/track
+78 tracks
+custom 5-to-8 GCR encoding (very similar to Apple II's 5-and-3 encoding)
+300 rpm
+FF FF FF used as sync-start, AB sync-mark for sector header, DE sync-mark for sector data
+FAT12 File System
+
+ROHM
+BA6580DK
+Read/Write Amplifier for FDD
+
+Emulation Status:
+Printer not working
+Floppy Disk writing not working
+Floppy Disk Sync not implemented (reading works)
+Dictionary ROM not working
+Cursor shapes not implemented except block cursor
+
+TODO: find self-test; verify RAM address map
+// 8kb SRAM, 64kb DRAM <- where?
+
+***************************************************************************/
+
+namespace {
+	constexpr uint16_t sync_table[]{
```

Comment:
This disk format should really go in src/lib/formats with the others.

Reply:
As @rb6502 says, please move the floppy format to src/lib/formats and add the glue to scripts/src/formats.lua so it can be used by the auxiliary tools.  Also, please use lowercase hex digits consistently.

Reply:
inside an if `opt_tool(FORMATS, "...") then` statement? how do I configure these? or in the main `files {` section?